### DISCUSSION TITLE
Is it possible to exclude test paths via CLI?

### DISCUSSION THREAD

**Discussion by freva** [vitest-dev/vitest#2113] [Q&A] [answered]
We have two sets of test: Simple JS test and slow visual tests. The simple test filenames match `*.test.js`, while the visual tests match `*.visual.test.js`. With `jest` we could have separate commands to run only one of the types, e.g.:
```
{
  "scripts": {
    "test": "CI=true jest --testPathIgnorePatterns=.*.visual.test.js$",
    "visual": "CI=true jest visual.test.js --runInBand --ci --forceExit --detectOpenHandles"
  },
  ...
}
```
From jest help:
```
      --testPathIgnorePatterns      An array of regexp pattern strings that are
                                    matched against all test paths before
                                    executing the test. If the test path matches
                                    any of the patterns, it will be skipped.
```
With `vitest` there is `exclude` in the config, but that seems to always exclude the files, even if I add them to `--testNamePattern`, and there doesn't seem to be any option to exclude tests with a command line argument.

**Comment by sheremet-va** [comment, marked as answer]
You can call vitest with custom env variable, and based on that alter config:

```ts
exclude: process.env.TEST_VISUAL ? defaultExclude : ["*.visual.test.js"]
```

**Reply by freva** [reply]
Thanks, that works.

**Comment by ildede** [comment]
After [`feat: add --exclude CLI flag` #4279](https://github.com/vitest-dev/vitest/pull/4279) it's also possible to use `--exclude` in CLI.
Example:
```
vitest --exclude "**/excluded-file"
```
